src/background/translate-persisted-data.ts

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { PersistedData } from 'background/get-persisted-data';
+import {
+    InjectingState,
+    VisualizationStoreData,
+} from 'common/types/store-data/visualization-store-data';
+import { forEach } from 'lodash';
+
+export function translatePersistedData(persistedData: PersistedData): PersistedData {
+    forEach(persistedData.tabData, tabData => {
+        translateInjectionState(tabData.visualizationStoreData);
+    });
+
+    return persistedData;
+}
+
+function translateInjectionState(data: VisualizationStoreData) {
+    if (data.injectingState !== undefined) {
+        // Preserve any existing injection state
+        return;
+    }
+
+    if (data['injectingStarted']) {
+        data.injectingState = InjectingState.injectingStarted;
+    } else {
+        data.injectingState = data['injectingRequested']
+            ? InjectingState.injectingRequested
+            : InjectingState.notInjecting;
+    }
+    delete data['injectingRequested'];
```

Comment:
In general, a function named `translate` that returns the same type it takes in shouldn't be mutating the original copy of the data - that's extremely misleading based on the name and an easy source of bugs. If mutating it makes sense for a given application, it makes it a lot harder to misunderstand the consuming code if both the name and typings are clear that it's mutating the original (eg, using a verb like `update` instead of `translate` and returning `void`).